python/tskit/drawing.py

```diff
@@ -617,73 +697,59 @@ def draw(self):
         # Iterate over nodes, adding groups to reflect the tree heirarchy
         stack = []
         for u in tree.roots:
-            grp = dwg.g(
-                class_=" ".join(self.info_classes(u)),
-                transform=f"translate({rnd(node_x_coord_map[u])} "
-                f"{rnd(node_y_coord_map[u])})",
+            stack.extend(
+                self.add_node(
+                    self.root_group, u, node_x_coord_map[u], node_y_coord_map[u],
+                )
             )
-            stack.append((u, self.root_group.add(grp)))
         while len(stack) > 0:
-            u, curr_svg_group = stack.pop()
-            pu = node_x_coord_map[u], node_y_coord_map[u]
-            for focal in tree.children(u):
-                fx = node_x_coord_map[focal] - pu[0]
-                fy = node_y_coord_map[focal] - pu[1]
-                new_svg_group = curr_svg_group.add(
-                    dwg.g(
-                        class_=" ".join(self.info_classes(focal)),
-                        transform=f"translate({rnd(fx)} {rnd(fy)})",
-                    )
-                )
-                stack.append((focal, new_svg_group))
-
-            o = (0, 0)
-            v = tree.parent(u)
+            curr = stack.pop()
+            u = curr.node
+            if curr.mutation is None:
+                # This is a tree node, not a mutation group
+                for c in tree.children(u):
+                    dx = node_x_coord_map[c] - node_x_coord_map[u]
+                    dy = node_y_coord_map[c] - node_y_coord_map[u]
+                    stack.extend(self.add_node(curr.g, c, dx, dy))
 
             # Add edge first => below
-            if v != NULL:
-                self.add_class(self.edge_attrs[u], "edge")
-                pv = node_x_coord_map[v], node_y_coord_map[v]
-                dx = pv[0] - pu[0]
-                dy = pv[1] - pu[1]
+            dx, dy = curr.edge_dxy
+            if dx == dy == 0:
```

Comment:
Ouch, I didn't even realise that python let you do this. ``if dx == 0 and dy == 0``

Reply:
I doubt this does what you expect, either, so we're probably missing some test coverage. This should mean (dx == dy) == 0 -> (True | False) == 0 .

Reply:
I'm pretty sure a==b==c does the expected thing due to [comparison operator chaining](https://docs.python.org/3/reference/expressions.html#comparisons); "unlike C, expressions like a < b < c have the interpretation that is conventional in mathematics"

So 0==0==1 is False, and all works as expected. Personally I find this syntax clearer, but then my head is rarely in C mode.

Reply:
I prefer `if dx == 0 and dy == 0` as it is unambiguous for both "C thinkers" and "mathematics thinkers".

Reply:
Sure, happy to switch: just pointing out that it's not unreasonable.

Reply:
(by the way, does that mean you also dislike `a<b<c`, which I fund much clearer than breaking up the test into 2 halves, like you have to do in C)?

Reply:
You're right, looks like ``0 == 0 == 0`` is True rather than False. But, it's not obvious, so I would be against using it. Stuff like ``a < b < c`` is great, on the other hand, as the meaning is unambiguous (to my eyes anyway).